### PR TITLE
Simplify code to calculate working dir of kernel & add tests 

### PR DESCRIPTION
**Pull request by DonJayamanne**
Part of #11552

### CODE REVIEW

File: src/platform/common/application/workspace.node.ts [hunk 1]
```diff
@@ -6,32 +6,27 @@ import { BaseWorkspaceService } from './workspace.base';
...
-    return workspace.getWorkspaceFolder(resource)?.uri.fsPath || workspace.rootFolder?.fsPath || process.cwd();
-}
 
 /**
  * Node implementation of the workspace service. Computing working directory is different for node.
  */
 @injectable()
 export class WorkspaceService extends BaseWorkspaceService {
-    public computeWorkingDirectory(resource: Resource): Promise<string> {
-        return computeWorkingDirectory(resource, this);
```
Comment (DonJayamanne) [inline src/platform/common/application/workspace.node.ts:35]: Not sure why we were doing this, passing `this` into a function.
Inlined the function here for improved readability

File: src/platform/common/utils.node.ts [hunk 2]
```diff
@@ -29,45 +23,3 @@ export async function tryGetRealPath(expectedPath: Uri): Promise<Uri> {
         return expectedPath;
     }
 }
-
-export async function calculateWorkingDirectory(
-    configService: IConfigurationService,
```
Comment (DonJayamanne) [inline src/platform/common/utils.node.ts:34]: This code doesn't belong here, this is very specific to kernels hence should not be under `platnform/common `

File: src/kernels/kernelStartupCodeProvider.node.ts [hunk 3]
```diff
@@ -2,80 +2,43 @@
...
-            isLocalConnection(kernel.kernelConnectionMetadata) ||
-            isLocalHostConnection(kernel.kernelConnectionMetadata)
-        ) {
-            let suggestedDir = await calculateWorkingDirectory(
-                this.configService,
-                this.workspaceService,
-                this.fs,
-                kernel.resourceUri
-            );
-            if (suggestedDir && (await this.fs.exists(suggestedDir))) {
```
Comment (DonJayamanne) [inline src/kernels/kernelStartupCodeProvider.node.ts:49]: As you can see we have more complex logic here and its partially split into two files.
With this PR its all in 1 place with tests.

File: .eslintrc.js [hunk 4]
```diff
@@ -266,7 +266,7 @@ module.exports = {
                         message: 'Importing test modules from ./src/test into extension code is not allowed.'
                     },
                     {
-                        target: './src/**[!test]**/**/*[!.node].ts',
+                        target: './src/**[!test]**/**/*[!.node|.unit].ts',
```
Comment (DonJayamanne) [inline .eslintrc.js:269]: Unit tests can import files from `*.node.ts` files for testing.

File: src/kernels/kernelStartupCodeProvider.node.ts [hunk 5]
```diff
@@ -2,80 +2,43 @@
...
-                        this.workspaceService,
-                        this.configService.getSettings(kernel.resourceUri)
-                    )
-                );
-                if (suggestedDir && (await this.fs.exists(suggestedDir))) {
-                    traceInfo('UpdateWorkingDirectoryAndPath in Kernel');
-                    return this.getChangeDirectoryCode(kernel, suggestedDir);
-                }
-            }
+        // If this is a live kernel, we shouldn't be changing anything by running startup code.
```
Comment (IanMatthewHuff) [inline src/kernels/kernelStartupCodeProvider.node.ts:34]: Comment might be in the wrong place now?

Comment (IanMatthewHuff) [reply src/kernels/kernelStartupCodeProvider.node.ts:34]: Small suggestion here, these checks are doubled up both here and in getWorkingDirectory. Maybe could just be in one place? If get working directory just returns undefined for remote or non-python then these checks here don't seem needed as a function prereq.

**Review by IanMatthewHuff — APPROVED** [review]
Looks good. Much cleaner to read now. Had one possible suggestion (looks like some error cases are checked twice).

File: src/kernels/kernelStartupCodeProvider.node.ts (hunk 5 above)
Comment (DonJayamanne) [reply src/kernels/kernelStartupCodeProvider.node.ts]: Agreed, Fixed

**Pull request merged**